tests/ServerTest.php

```diff
@@ -1023,6 +1023,162 @@ function ($data) use (&$buffer) {
         $this->assertInstanceOf('InvalidArgumentException', $error);
     }
 
+    public function testInvalidChunkHeaderResultsInErrorResponse()
```

Comment:
Changed the test names.
I consider to reorganize it in a seprated PR. Currently the `ServerTest` is more an integration test than an unit test. That is completly fine but maybe this should be seperated. I would keep the tests in this PR, because this would affect several other tests too.